src/sentry/rules/processor.py

```diff
@@ -115,30 +110,33 @@ def apply_rule(self, rule):
             else:
                 filter_list.append(rule_cond)
 
-        if not condition_list:
-            return
-
-        condition_iter = (self.condition_matches(c, state, rule) for c in condition_list)
-        filter_iter = (self.condition_matches(f, state, rule) for f in filter_list)
+        # if conditions exist evaluate them, otherwise move to the filters section
```

Comment:
previously having no conditions would mean the alert rule would not trigger, this was probably impossible due to validation though. However, if there are no conditions we actually want the rule to pass as conditions are optional now.

Reply:
I thought conditions were required and filters were optional?

Reply:
yea that was the case before but now with the introduction of filters we want to give the option to have a rule with only alert filters (no conditions). Previously the only way to do this was to add a `The event occurs` condition to the set of conditions and then add the filters you want. However, this is redundant as every time an alert rule is evaluated it's already implied that an event has occurred. Therefore we removed `The event occurs` condition and made conditions optional.

Reply:
Does this mean that it's no longer valid to use the `The event occurs` condition on its own? Or if the user enters no conditions/filters do we just fire on all events?